### ISSUE TITLE
RemoteServiceException: CannotPostForegroundServiceNotificationException

### ISSUE THREAD

**Issue by AjeshRPai** [wordpress-mobile/WordPress-Android#18495]
Crash: Sentry link 
```
android.app.RemoteServiceException$CannotPostForegroundServiceNotificationException: Bad notification for startForeground
    at android.app.ActivityThread.throwRemoteServiceException(ActivityThread.java:2222)
    at android.app.ActivityThread.-$$Nest$mthrowRemoteServiceException
    at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2508)
    at android.os.Handler.dispatchMessage(Handler.java:106)
    at android.os.Looper.loopOnce(Looper.java:226)
    at android.os.Looper.loop(Looper.java:313)
    at android.app.ActivityThread.main(ActivityThread.java:8757)
    at java.lang.reflect.Method.invoke(Method.java)
    at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:571)
    at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1067)
```

**Comment by AjeshRPai** [comment]
Hey @irfano 

I noticed this issue in Sentry and its happening only on Android 13. Since you were the DRI for Android 13 migration, i figured you might have more context or knowledge about this crash. Please feel free to un-assign if you feel this is not related to Android 13/ its an already known issue. 

Thanks in Advance.

**Comment by irfano** [comment]
Hello, @AjeshRPai! I was AFK for the entire week when you messaged me. Unfortunately, I missed the notification when I returned. I will look into this issue this week.

**Comment by irfano** [comment]
This exception tells us we are trying to show notifications without creating a notification channel.

This issue was happening on PhaseFour, PhaseNewUsers, and PhaseSelfHostedUsers in the migration because notifications were disabled in these phases. So, we did not create notification channels, yet we were attempting to show media upload notifications even though we didn't create notification channels.

I'm closing the issue without any development because we're already in the last phase. The next step will be removing all migration settings and this issue won't occur again. 

_There are still some new crashes coming recently, but not many. They may have stale, cached remote flags._

**To reproduce:**
_(We need to manipulate feature flag_
1. In Android Studio, update [shouldShowNotifications()](https://github.com/wordpress-mobile/WordPress-Android/blob/996b10c87a5486ebc3140633dd094d3a82a02bf6/WordPress/src/main/java/org/wordpress/android/ui/jetpackoverlay/JetpackFeatureRemovalPhaseHelper.kt#L135-L141). It should return false to mimic PhaseFour, PhaseNewUsers, and PhaseSelfHostedUsers.
2. Make a clean install of the WP app.
3. Log in.
4. Create a post.
5. Add a video block and a video.
6. Notice it will crash when the video is being uploaded.

Sentry link: https://a8c.sentry.io/issues/3909129591/?query=CannotPostForegroundServiceNotificationException&referrer=issue-stream&statsPeriod=90d&stream_index=0

**Comment by derekblank** [comment]
> There are still some new crashes coming recently, but not many. They may have stale, cached remote flags.

Hey @irfano 👋 We're still receiving [Sentry event reports for this in 23.6](https://a8c.sentry.io/issues/3909129591/events/?query=release%3A%2223.6%22), but it looks like they are from the same 3 - 4 users. Given the comment above, I just wanted to double check if this is expected or not, as it's currently the third-highest WPAndroid crash.

**Comment by irfano** [comment]
I have shared how I reproduced this issue previously (https://github.com/wordpress-mobile/WordPress-Android/issues/18495#issuecomment-1596227519). However, it appears that Sentry crashes do not come from that case.
This crash always occurs after the log `i/WordPress-MAIN: UploadService > Destroyed`. Following the destruction of `UploadService`, `[UploadService.startForeground()](https://github.com/wordpress-mobile/WordPress-Android/blob/trunk/WordPress/src/main/java/org/wordpress/android/ui/uploads/PostUploadNotifier.java#L138)` is invoked, and leading to the crash. The underlying issue lies in the destruction of `UploadService`, which occurs subsequent to an upload error.

So, I believe the root cause of this crash is a media upload issue that results in the destruction of `UploadService`. I was unable to reproduce the issue, but I anticipate that **fixing media upload issues** and **migrating `UploadService` to Workmanager** will address this crash. Internal issue: https://github.com/Automattic/jetpack-mobile-roadmap/issues/1

I am reopening the issue as it's still active.

**Comment by antonis** [comment]
> This crash always occurs after the log i/WordPress-MAIN: UploadService > Destroyed. Following the destruction of UploadService, [UploadService.startForeground()](https://github.com/wordpress-mobile/WordPress-Android/blob/trunk/WordPress/src/main/java/org/wordpress/android/ui/uploads/PostUploadNotifier.java#L138) is invoked, and leading to the crash. The underlying issue lies in the destruction of UploadService, which occurs subsequent to an upload error.

Based on this I believe that the UploadService fix might have covered this too https://github.com/wordpress-mobile/WordPress-Android/pull/19987
I'll monitor `24.1+` to verify if this is the case or there is a regression.

**Comment by antonis** [comment]
> Based on this I believe that the UploadService fix might have covered this too https://github.com/wordpress-mobile/WordPress-Android/pull/19987
I'll monitor 24.1+ to verify if this is the case or there is a regression.

Unfortunately this is not the case :( The issue persist in `24.1` and 11 new crashes of [WORDPRESS-ANDROID-2H8Q](https://a8c.sentry.io/issues/3909129591/?referrer=github_integration) in this version were recorded.

**Comment by antonis** [comment]
> Unfortunately this is not the case :( The issue persist in 24.1 and 11 new crashes of [WORDPRESS-ANDROID-2H8Q](https://a8c.sentry.io/issues/3909129591/?referrer=github_integration) in this version were recorded.

At the time of writing [11 crashes were recorder in `24.1` by 2 users](https://a8c.sentry.io/issues/3909129591/events/?project=5731682&project=1438088&query=release%3A24.1%2A&referrer=issue-stream&statsPeriod=90d&stream_index=0) all in the WordPress app:


> I have shared how I reproduced this issue previously (https://github.com/wordpress-mobile/WordPress-Android/issues/18495#issuecomment-1596227519). However, it appears that Sentry crashes do not come from that case.

I was able to reproduce this type of crash only with the steps @irfano provided above.

> So, we did not create notification channels, yet we were attempting to show media upload notifications even though we didn't create notification channels.

Given that I could not find any other reproducible case for the error I proceeded with a fix covering this case assuming that on some edge cases (e.g. the client failed to load the latest migration phase from the server) this might still occur.

On the stack trace two `No Channel found` errors occur for the transient and normal channels just berore the crash:
```
No Channel found for pkg=org.wordpress.android.beta, channelId=wpandroid_notification_transient_channel_id, id=369571377, tag=null, opPkg=org.wordpress.android.beta, callingUid=10659, userId=0, incomingUserId=0, notificationUid=10659, notification=Notification(channel=wpandroid_notification_transient_channel_id shortcut=null contentView=null vibrate=null sound=null defaults=0x0 flags=0x8 color=0xff0675c4 vis=PRIVATE)
...
No Channel found for pkg=org.wordpress.android.beta, channelId=wpandroid_notification_normal_channel_id, id=2, tag=null, opPkg=org.wordpress.android.beta, callingUid=10659, userId=0, incomingUserId=0, notificationUid=10659, notification=Notification(channel=wpandroid_notification_normal_channel_id shortcut=null contentView=null vibrate=null sound=null defaults=0x0 flags=0x18 color=0xffc9356e actions=1 vis=PRIVATE)
FATAL EXCEPTION: main
Process: org.wordpress.android.beta, PID: 12180
android.app.RemoteServiceException$CannotPostForegroundServiceNotificationException: Bad notification for startForeground
	at android.app.ActivityThread.throwRemoteServiceException(ActivityThread.java:2078)
	at android.app.ActivityThread.-$$Nest$mthrowRemoteServiceException(Unknown Source:0)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2369)
	at android.os.Handler.dispatchMessage(Handler.java:106)
	at android.os.Looper.loopOnce(Looper.java:205)
	at android.os.Looper.loop(Looper.java:294)
	at android.app.ActivityThread.main(ActivityThread.java:8177)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:552)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:971)
```
By experimenting I defined that only the transient channel is necessary to prevent this crash and prepared a fix in https://github.com/wordpress-mobile/WordPress-Android/pull/20168

**Comment by antonis** [comment]
Marking the issue as resolved in `24.2` with https://github.com/wordpress-mobile/WordPress-Android/pull/20168. We can reopen if the issue reoccurs.
